src/CodeGen_C.cpp

```diff
@@ -1235,9 +1250,19 @@ void CodeGen_C::visit(const Call *op) {
             rhs << "_ucon, ";
         }
 
+        // TODO: special-case for https://github.com/halide/Halide/issues/2099;
+        // embedded buffers will generate code for bounds query, which can
+        // never execute (_halide_buffer_is_bounds_query() always returns false
+        // for them), but the generated bounds-query code won't compile due to 
+        // const correctness. We add some const_casts here as a temporary
+        // workaround.
```

Comment:
I don't think this workaround will be so easy to resolve. If embedded buffers in one pipeline are passed to another pipeline (e.g. for an extern stage), the buffer objects need to be non-const, because the constness (from the POV of the extern stage) is a runtime property. Seems simpler to just declare embedded buffers as non-const.